components/server/src/gitlab/gitlab-repository-provider.ts

```diff
@@ -140,8 +140,29 @@ export class GitlabRepositoryProvider implements RepositoryProvider {
         return result.slice(1).map((c: GitLab.Commit) => c.id);
     }
 
-    // TODO: implement repo search
     public async searchRepos(user: User, searchString: string): Promise<RepositoryInfo[]> {
-        return [];
+        const logCtx = { userId: user.id };
+
+        try {
+            const result = await this.gitlab.run<GitLab.Project[]>(user, async (gitlab) => {
+                return gitlab.Projects.all({
+                    membership: true,
+                    search: searchString,
+                    simple: true,
+                });
+            });
+
+            if (GitLab.ApiError.is(result)) {
+                throw result;
+            }
+
+            return result.map((result) => {
+                return { url: result.web_url, name: result.name };
+            });
+        } catch (e) {
+            log.warn(logCtx, "Error searching repos", e);
+
+            throw e;
+        }
```

Comment:
This is what I ended up doing by mimicking the GitHub code. Not sure what the right approach is tho
https://github.com/gitpod-io/gitpod/blob/cf4f4687e601b3cd2e68921bbf7a14e97fb0e1d7/components/server/src/github/github-repository-provider.ts#L275-L279

Reply:
Ah, I think we could remove that try/catch here and in the github searchRepos fn (I meant to do that after doing bitbucket fns), and let the caller handle the catch/logging, which it does already in `gitpod-server-impl:searchRepositories`.